sopel/plugins/handlers.py

```diff
@@ -348,7 +356,19 @@ def has_setup(self):
         """
         return hasattr(self._module, 'setup')
 
-    def register(self, bot):
+    def get_capability_requests(self) -> List[plugin_decorators.capability]:
+        return [
+            module_attribute
+            for module_attribute in vars(self._module).values()
+            if isinstance(module_attribute, plugin_decorators.capability)
+        ]
+
+    def register(self, bot: Sopel) -> None:
+        # capabilities are directly registered
+        for cap_request in self.get_capability_requests():
+            bot.cap_requests.register(self.name, cap_request)
+
+        # plugin callables go through ``bot.add_plugin``
```

Comment:
Busted: this is absolutely a not-so-secret way to migrate from the old-style `bot.add_plugin` method to a version where the Plugin handler exposes properly its plugin callables (rules, commands, jobs, URLs, and now CAP requests).

Reply:
Maybe annotate `bot.add_plugin` with a TODO to that effect (that it's intended to be replaced)?

I think it's great for the plugin handler classes to take care of collecting the relevant callables, eventually. Just important to remember that `bot.add_plugin` is _technically_ part of a public API, and therefore it needs to be marked as deprecated (or otherwise flagged as "will change") well in advance.